scipy/ndimage/morphology.py

```diff
@@ -257,11 +253,7 @@ def _binary_erosion(input, structure, iterations, mask, output,
             if not structure.shape[ii] & 1:
                 origin[ii] -= 1
         if mask is not None:
-            msk = numpy.asarray(mask)
```

Comment:
This whole branch made very little sense: calling `.astype()` always makes a copy, so the check for `msk is mask` was completely unnecessary. It is now possible that mask is the original array, not a copy, but `binary_erosion2()` never writes to mask, only reads it, and it will make a copy if the array is not aligned and contiguous.